src/sentry/workflow_engine/models/data_condition.py

```diff
@@ -57,8 +57,24 @@ class DataCondition(DefaultFieldsModel):
         on_delete=models.CASCADE,
     )
 
-    def evaluate_value(self, value: float | int) -> DetectorPriorityLevel | None:
-        # Note: We'll have other types other than int/DetectorPriorityLevel here, keeping it simple for now
+    def get_condition_result(self) -> DataConditionResult:
+        match self.condition_result:
+            case float() | bool():
```

Comment:
I think that this will always prevent the `int() | DetectorPriorityLevel()` case right? Since all ints can be floats.

Do we need to embed more information about the type in the json maybe?

Reply:
🤔 Hmm - i do have a test to confirm that this doesn't happen -- i pass a float and ensure it's a float response, and same with an int. I think it's related to the `class patterns` check i linked to in another comment. 🪄